spk/owncloud/src/service-setup.sh

```diff
@@ -0,0 +1,507 @@
+        CFG_UPDATE="no"
+        # Check if Apache is the selected back-end
+        if [ ! "$WS_BACKEND" = "2" ]; then
+            echo "Set Apache as the back-end server"
+            ${JQ} '.default.backend = 2' ${WS_CFG_PATH}/${WS_CFG_FILE} > ${TEMPDIR}/${WS_CFG_FILE}
+            ${MV} ${WS_CFG_PATH}/${WS_CFG_FILE} ${WS_CFG_PATH}/${WS_CFG_FILE}.bak
+            rsync -aX ${TEMPDIR}/${WS_CFG_FILE} ${WS_CFG_PATH}/ 2>&1
+            ${RM} ${TEMPDIR}/${WS_CFG_FILE}
+            CFG_UPDATE="yes"
+        fi
+        # Check if default PHP profile is selected
+        if [ -z "$WS_PHP" ] || [ "$WS_PHP" = "null" ]; then
+            echo "Enable default PHP profile"
+            # Locate default PHP profile
+            PHP_PROF_ID="$(${JQ} -r '. | to_entries[] | select(.value | type == "object" and .profile_desc == "'"$PHP_PROF_NAME"'") | .key' "${WS_CFG_PATH}/${PHP_CFG_FILE}")"
+            ${JQ} ".default.php = \"$PHP_PROF_ID\"" "${WS_CFG_PATH}/${WS_CFG_FILE}" > ${TEMPDIR}/${WS_CFG_FILE}
+            ${MV} ${WS_CFG_PATH}/${WS_CFG_FILE} ${WS_CFG_PATH}/${WS_CFG_FILE}.bak
+            rsync -aX ${TEMPDIR}/${WS_CFG_FILE} ${WS_CFG_PATH}/ 2>&1
+            ${RM} ${TEMPDIR}/${WS_CFG_FILE}
+            CFG_UPDATE="yes"
+        fi
+        # Check for ownCloud PHP profile
+        if ! ${JQ} -e '.["com-synocommunity-packages-owncloud"]' "${WS_CFG_PATH}/${PHP_CFG_FILE}" >/dev/null; then
+            echo "Add PHP profile for ownCloud"
+            ${JQ} --argfile ocNode ${SYNOPKG_PKGDEST}/web/owncloud.json '.["com-synocommunity-packages-owncloud"] = $ocNode' ${WS_CFG_PATH}/${PHP_CFG_FILE} > ${TEMPDIR}/${PHP_CFG_FILE}
+            ${MV} ${WS_CFG_PATH}/${PHP_CFG_FILE} ${WS_CFG_PATH}/${PHP_CFG_FILE}.bak
+            rsync -aX ${TEMPDIR}/${PHP_CFG_FILE} ${WS_CFG_PATH}/ 2>&1
+            ${RM} ${TEMPDIR}/${PHP_CFG_FILE}
+            CFG_UPDATE="yes"
+        fi
+        # Check for updated PHP template
+        if grep -q -E '^(user|listen\.owner) = http$' "${WS_TMPL_PATH}/${WS_TMPL_FILE}"; then
+            echo "Update PHP template for ownCloud"
+            rsync -aX ${WS_TMPL_PATH}/${WS_TMPL_FILE} ${TEMPDIR}/ 2>&1
+            SUBST_TEXT="{{#fpm_settings.user_owncloud}}sc-owncloud{{/fpm_settings.user_owncloud}}{{^fpm_settings.user_owncloud}}http{{/fpm_settings.user_owncloud}}"
+            ${SED} -i "s|^user = http$|user = ${SUBST_TEXT}|g; s|^listen.owner = http$|listen.owner = ${SUBST_TEXT}|g" "${TEMPDIR}/${WS_TMPL_FILE}"
+            ${MV} ${WS_TMPL_PATH}/${WS_TMPL_FILE} ${WS_TMPL_PATH}/${WS_TMPL_FILE}.bak
+            rsync -aX ${TEMPDIR}/${WS_TMPL_FILE} ${WS_TMPL_PATH}/ 2>&1
+            ${RM} ${TEMPDIR}/${WS_TMPL_FILE}
+            CFG_UPDATE="yes"
+        fi
+        # Check for ownCloud Apache config
+        if [ ! -f "/usr/local/etc/apache24/sites-enabled/owncloud.conf" ]; then
+            echo "Add Apache config for ownCloud"
+            rsync -aX ${SYNOPKG_PKGDEST}/web/owncloud.conf /usr/local/etc/apache24/sites-enabled/ 2>&1
```

Comment:
I understand that rsync is powerful but how is that relevant compared to a plain old cp ?

Reply:
I didn't pick it for any specific reason, just seen `rsync` being used much more in other packages to copy files. In this case the `-aX` will preserve attributes and extended attributes when copying files.

Reply:
Which is equivalent to `cp --preserve=mode,ownership,timestamps,xattr` but I give you that it does the trick then :)